registry-v2.md

```diff
@@ -0,0 +1,82 @@
+# Registry v2
+
+*NOTE: This registry is not yet deployed on hex.pm*
+
+The first version of the registry format used a single file for the entire registry. While this was efficient in the number of HTTP requests needed to resolve dependencies, the size of the registry would grow out of control as the repository grows.
+
+The new version of the registry format is split into multiple files to make it scale as the repository grows with more packages.
+
+For every package in the repository there is a package registry file under `/packages/[NAME]` in the repository, it contains all the releases of that package and all dependencies of the releases. There are also two index files `/names` and `/versions` that contains all package names and all package names+versions respectively.
+
```

Comment:
Ideally it should be the latter. This document only outlines the format of the resources and how they should be located (the path). The `endpoints.md` document specifies how the resources can be accessed over HTTP, but it can easily be extended to other path based protocols, such as FTP or even direct filesystem access.

With that in mind, I should make sure we do not mention any storage formats implicitly, such as "file".